src/hooks/useAddTokenToMetamask.ts

```diff
@@ -33,7 +33,9 @@ export default function useAddTokenToMetamask(token: Token | undefined): {
             },
           },
         })
-        .then((success) => setSuccess(success))
+        .then((success: SetStateAction<boolean | undefined>) =>
```

Comment:
This doesn't seem correct - `success` is just a boolean and `SetStateAction` is the type for `setState` in the context `const [state, setState] = useState()`

`type SetStateAction<S> = S | ((prevState: S) => S);`